EIPS/eip-7069.md

```diff
@@ -50,31 +50,32 @@ Lastly, the introduction of the `RETURNDATA*` instructions ([EIP-211](./eip-211.
 
 We introduce four new instructions:
 
-- `CALL2` (`0xf8`) with arguments `(target_address, input_offset, input_size, value)` 
-- `DELEGATECALL2` (`0xf9`) with arguments `(target_address, input_offset, input_size)`
-- `STATICCALL2` (`0xfb`) with arguments `(target_address, input_offset, input_size)`
+- `EXTCALL` (`0xf8`) with arguments `(target_address, input_offset, input_size, value)` 
+- `EXTDCALL` (`0xf9`) with arguments `(target_address, input_offset, input_size)` - delegate call
+- `EXTSCALL` (`0xfb`) with arguments `(target_address, input_offset, input_size)` - static call
 - `RETURNDATALOAD` (`0xf7`) with argument `offset`
 
-Execution semantics of `*CALL2`:
+Execution semantics of `EXT*CALL`:
 
 1. Charge `WARM_STORAGE_READ_COST` (100) gas.
 2. Pop required arguments from stack, fail with error on stack underflow.
-4. If `value` is non-zero:
+3. If `value` is non-zero:
     3a. Fail with error if the current frame is in `static-mode`.
-    3b. Fail with error if the balance of the current account is less than `value`.
-    3c. Charge `CALL_VALUE_COST` gas.
+    3b. Charge `CALL_VALUE_COST` gas.
 4. Peform (and charge for) memory expansion using `[input_offset, input_size]`.
 5. If `target_address` is not in the `warm_account_list`, charge `COLD_ACCOUNT_ACCESS - WARM_STORAGE_READ_COST` (2500) gas.
 6. If `target_address` is not in the state and the call configuration would result in account creation, charge `ACCOUNT_CREATION_COST` (25000) gas.
     - The only such case in this EIP is if `value` is non-zero.
-7. Calculate the gas available to callee as caller's remaining gas reduced by `max(ceil(gas/64), MIN_RETAINED_GAS)` (`MIN_RETAINED_GAS` is 5000).
-8. Fail with error if the gas available to callee at this point is less than `MIN_CALLEE_GAS` (2300).
-9. Perform the call with the available gas and configuration.
-10. Push a status code on the stack:
-    11a. `0` if the call was successful.
-    11b. `1` if the call has reverted.
-    11c. `2` if the call has failed.
-11. Gas not used by the callee is returned to the caller.
+7. Calculate the gas available to callee as caller's remaining gas reduced by `max(floor(gas/64), MIN_RETAINED_GAS)`.
+8. Fail with error if the gas available to callee at this point is less than `MIN_CALLEE_GAS`.
+9. If `value` is non-zero, fail lightly if the balance of the current account is less than `value`.
+10. Fail lightly if call stack depth equals `1024`.
```

Comment:
What does "fail Lightly" mean?  dose "Fail with error" mean revert and take all available gas and "fail lightly" mean fail consuming only already charged gas?

Reply:
yes (except "Fail with error" is not a revert but an exceptional abort, I presume this was the original intention), I'll specify these.

But you've uncovered an interesting issue - what should be pushed to stack in this case? For a light failure in EOFCREATE/TXCREATE we return `0` (with a different meaning), same for CREATE/CREATE2, but for revamped calls that would mean success.

I also see that `evmone` would return `0`, so making a self note to amend this (I'm working on that now in parallel).

Reply:
Proposing a resolution (`3` on stack for light failure) https://github.com/ethereum/EIPs/pull/8287/commits/54f6dcd92c95a21841e68fd7888dab5e02f6f51a to push the discussion forward. WDYT?

Reply:
https://github.com/ethereum/EIPs/pull/8287/commits/477e269900f21a4cdd6f48dfe08f36a8a750af79 adheres to what we spoke about on the EOF implementers call - `1` as status code for light failure